src/shared/types/form.ts

```diff
@@ -0,0 +1,19 @@
+// This type defines the props (name and control) for the hookform version of form inputs.
+
+import { Control, FieldValues, Path } from "react-hook-form";
+
+// When control is provided, onChange and onBlur should not be allowed.
+export interface WithHookForm<U extends FieldValues> {
+  name: Path<U>;
+  control: Control<U>;
+  onChange?: never;
+  onBlur?: never;
+}
+// This type allows onChange and onBlur to be passed when there is no control.
+// Also makes name optional and a generic string instead of a form Path.
+export interface WithoutHookForm<T> {
```

Comment:
Semantics, but both have hook forms; the difference is when they are controlled or uncontrolled.

Reply:
Not sure I understand this, if I don't pass `control` to the component then hook form should have no idea that particular field exists. I am using `useController` inside on either case, is that what you meant?

These names makes sense to me but I'm open to change them to whatever seems clearer.